packages/cardpay-reward-programs/cardpay_reward_programs/rule.py

```diff
@@ -0,0 +1,90 @@
+import hashlib
+import json
+from abc import ABC, abstractmethod
+
+import duckdb
+import pandas as pd
+from cardpay_reward_programs.utils import get_files
+
+
+class Rule(ABC):
+    """
+    A single image should run only a single rule
+    """
+
+    def __init__(self, core_parameters, user_defined_parameters):
+        self.set_core_parmaters(**core_parameters)
+        self.set_user_defined_parameters(**user_defined_parameters)
+
+    def set_core_parmaters(
+        self, subgraph_config_location, payment_cycle_length, valid_from, valid_to, token
+    ):
+        self.subgraph_config_location = subgraph_config_location
```

Comment:
I've been thinking on this and I think the config location is also not a core param. You could read from more than one.

Reply:
agreed. this was actually my thought until I reverted my changes from the module pr. Thanks.